src/gc/impl/manual/gc.d

```diff
@@ -0,0 +1,305 @@
+            onOutOfMemoryError();
+
+        instance = cast(ManualGC) memcpy(p, typeid(ManualGC).initializer.ptr,
+            typeid(ManualGC).initializer.length);
+
+        instance.__ctor();
+
+        gc = instance;
+    }
+
+    static void finalize()
+    {
+        if (instance !is null)
+        {
+            instance.Dtor();
+            cstdlib.free(cast(void*) instance);
+        }
+    }
+
+    this()
+    {
+    }
+
+    void Dtor()
+    {
+        cstdlib.free(roots);
+        cstdlib.free(ranges);
+    }
+
+    void enable()
+    {
+    }
+
+    void disable()
+    {
+    }
+
+    void collect() nothrow
+    {
+    }
+
+    void collectNoStack() nothrow
+    {
+    }
+
+    void minimize() nothrow
+    {
+    }
+
+    uint getAttr(void* p) nothrow
+    {
+        return 0;
+    }
+
+    uint setAttr(void* p, uint mask) nothrow
+    {
+        return 0;
+    }
+
+    uint clrAttr(void* p, uint mask) nothrow
+    {
+        return 0;
+    }
+
+    void* malloc(size_t size, uint bits, const TypeInfo ti) nothrow
+    {
+        void* p = cstdlib.malloc(size);
+
+        if (size && p is null)
+            onOutOfMemoryError();
+        return p;
+    }
+
+    BlkInfo qalloc(size_t size, uint bits, const TypeInfo ti) nothrow
+    {
+        BlkInfo retval;
+        retval.base = malloc(size, bits, ti);
+        retval.size = size;
+        retval.attr = bits;
+        return retval;
+    }
+
+    void* calloc(size_t size, uint bits, const TypeInfo ti) nothrow
+    {
+        void* p = cstdlib.calloc(1, size);
+
+        if (size && p is null)
+            onOutOfMemoryError();
+        return p;
+    }
+
+    void* realloc(void* p, size_t size, uint bits, const TypeInfo ti) nothrow
+    {
+        p = cstdlib.realloc(p, size);
+
+        if (size && p is null)
+            onOutOfMemoryError();
+        return p;
+    }
+
+    size_t extend(void* p, size_t minsize, size_t maxsize, const TypeInfo ti) nothrow
+    {
+        return 0;
+    }
+
+    size_t reserve(size_t size) nothrow
+    {
+        return 0;
+    }
+
+    void free(void* p) nothrow
+    {
+        cstdlib.free(p);
+    }
+
+    /**
+     * Determine the base address of the block containing p.  If p is not a gc
+     * allocated pointer, return null.
+     */
+    void* addrOf(void* p) nothrow
+    {
+        return null;
+    }
+
+    /**
+     * Determine the allocated size of pointer p.  If p is an interior pointer
+     * or not a gc allocated pointer, return 0.
+     */
+    size_t sizeOf(void* p) nothrow
+    {
+        return 0;
+    }
+
+    /**
+     * Determine the base address of the block containing p.  If p is not a gc
+     * allocated pointer, return null.
+     */
+    BlkInfo query(void* p) nothrow
+    {
+        return BlkInfo.init;
+    }
+
+    GCStats stats() nothrow
+    {
+        return GCStats.init;
+    }
+
+    void addRoot(void* p) nothrow @nogc
+    {
+        Root* r = cast(Root*) cstdlib.realloc(roots, (nroots + 1) * roots[0].sizeof);
```

Comment:
I know, I was just pointing that it would be better to reuse an existing container implementation, because it would be easier, should minimize the probability of bugs and it maybe faster.